### PR TITLE
[DNM] New Cluster-TLS API

### PR DESCRIPTION
**Pull request by weekface**
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

**Comment by tennix** [comment]
Is this API compatible with cert-manager?

**Comment by weekface** [comment]
We can add cert-manager support by add a new key:

```
tlsCluster:
  enabled: true
  secretName: "secretName"
  autoGenerated: {}
  certManager: {}
```

**Comment by tennix** [comment]
> We can add cert-manager support by add a new key:
> 
> ```
> tlsCluster:
>   enabled: true
>   secretName: "secretName"
>   autoGenerated: {}
>   certManager: {}
> ```

OK, sounds good to me.

File: charts/tidb-cluster/values.yaml [hunk 1]
```diff
@@ -55,10 +55,35 @@ discovery:
...
+  # #   2. Then create the secret with the following command:
+  # #   kubectl create secret generic <secret-name> --namespace=<namespace>
+  # #       --from-file=pd-cert=<pd server certificate file path>
+  # #       --from-file=pd-key=<pd server key file path>
+  # #       --from-file=tikv-cert=<tikv server certificate file path>
+  # #       --from-file=tikv-key=<tikv server key file path>
+  # #       --from-file=tidb-cert=<tidb server certificate file path>
+  # #       --from-file=tidb-key=<tidb server key file path>
+  # #       --from-file=ca=<ca file path>
+  # secretName: "demo-tidb-components-secret"
```
Comment (cofyc) [inline charts/tidb-cluster/values.yaml]: - one secret for one component
- for serects, I suggest using dedicated tls secret generation method in Kubernetes, `kubectl create secret tls <secret>`

the reasons are:
- automation tool like cert-manager manages one secret in each Certificate CRD 
- we should following k8s tls secret format, then third party generated secrets can be used directly

Comment (weekface) [reply charts/tidb-cluster/values.yaml]: We are considering using one certificate for all tidb components. If we can, just need one `secret`.

Comment (weekface) [reply charts/tidb-cluster/values.yaml]: Or TiDB core Components(include pd tikv and tidb) use one single cert, and TiDB Binlog(include pump and drainer) use another cert. It’s the same for future TiFlash and other components...

Comment (weekface) [reply charts/tidb-cluster/values.yaml]: @cofyc The `kubectl create secret tls` only have `--cert` and `--key`, but without `--ca` flag, so we can't use it to create the user-defined certificate secret?

Comment (cofyc) [reply charts/tidb-cluster/values.yaml]: yes, we still need to use the `generic` type in this case.

**Comment by weekface** [comment]
comments addressed PTAL @cofyc @AstroProfundis @tennix

File: pkg/apis/pingcap/v1alpha1/tidbcluster.go [hunk 2]
```diff
@@ -299,11 +299,7 @@ func (tc *TidbCluster) GetClusterID() string {
 }
 
 func (tc *TidbCluster) IsTLSClusterEnabled() bool {
-	enableTLCluster := tc.Spec.EnableTLSCluster
-	if enableTLCluster == nil {
-		return defaultEnableTLSCluster
-	}
-	return *enableTLCluster
+	return tc.Spec.TLSCluster != nil && tc.Spec.TLSCluster.Enabled
```
Comment (Yisaer) [inline pkg/apis/pingcap/v1alpha1/tidbcluster.go]: It seems we are no longer using the `tc.Spec.EnableTLSCluster`, do we need to remove this property or set it as deprecated?

Comment (weekface) [reply pkg/apis/pingcap/v1alpha1/tidbcluster.go]: removed

**Comment by weekface** [comment]
Open a new PR instead: https://github.com/pingcap/tidb-operator/pull/1870

**Pull request closed without merging**